sdk/src/main/java/io/opentelemetry/sdk/trace/export/BatchSpansProcessor.java

```diff
@@ -411,4 +324,277 @@ public void run() {
+        return this;
+      }
+
+      /**
+       * Sets the maximum time an exporter will be allowed to run before being cancelled.
+       *
+       * <p>Default value is {@code 30000}ms
+       *
+       * @param exporterTimeoutMillis the timeout for exports in milliseconds.
+       * @return this
+       */
+      public Builder setExporterTimeoutMillis(int exporterTimeoutMillis) {
+        Utils.checkArgument(
+            exporterTimeoutMillis >= 0, "exporterTimeoutMillis must greater than or equal 0.");
+        this.exporterTimeoutMillis = exporterTimeoutMillis;
+        return this;
+      }
+
+      /**
+       * Sets the maximum number of Spans that are kept in the queue before start dropping.
+       *
+       * <p>See the BatchSampledSpansProcessor class description for a high-level design description
+       * of this class.
+       *
+       * <p>Default value is {@code 2048}.
+       *
+       * @param maxQueueSize the maximum number of Spans that are kept in the queue before start
+       *     dropping.
+       * @return this.
+       */
+      public Builder setMaxQueueSize(int maxQueueSize) {
+        Utils.checkArgument(maxQueueSize > 0, "maxQueueSize must be positive.");
+        this.maxQueueSize = maxQueueSize;
+        return this;
+      }
+
+      /**
+       * Sets the maximum batch size for every export. This must be smaller or equal to {@code
+       * maxQueuedSpans}.
+       *
+       * <p>Default value is {@code 512}.
+       *
+       * @param maxExportBatchSize the maximum batch size for every export.
+       * @return this.
+       */
+      public Builder setMaxExportBatchSize(int maxExportBatchSize) {
+        Utils.checkArgument(maxExportBatchSize > 0, "maxExportBatchSize must be positive.");
+        this.maxExportBatchSize = maxExportBatchSize;
+        return this;
+      }
+
+      private static boolean getProperty(
+          String name, Map<String, String> map, boolean defaultValue) {
+        if (map.containsKey(name)) {
+          return Boolean.parseBoolean(map.get(name));
+        }
+        return defaultValue;
+      }
+
+      private static int getProperty(String name, Map<String, String> map, int defaultValue) {
+        try {
+          return Integer.parseInt(map.get(name));
+        } catch (NumberFormatException ex) {
+          return defaultValue;
+        }
+      }
+
+      private static long getProperty(String name, Map<String, String> map, long defaultValue) {
+        try {
+          return Long.parseLong(map.get(name));
+        } catch (NumberFormatException ex) {
+          return defaultValue;
+        }
+      }
+
+      /**
+       * Builds the {@link Config} object.
+       *
+       * @return the {@link Config} object.
+       */
+      public Config build() {
+        return new AutoValue_BatchSpansProcessor_Config(
+            sampled, scheduleDelayMillis, maxQueueSize, maxExportBatchSize, exporterTimeoutMillis);
+      }
+    }
+  }
```

Comment:
this class is now over 600 lines long. I think it might be time to break out the Config and the Builder into their own files.

Reply:
not sure why you resolved this with no response, @thisthat

Reply:
Sorry, I forgot to comment on the decision taken in the SIG call.
Refactoring will be done in a separate PR when the configuration for other parts of OTel will be provided.

Reply:
ah, ok. I didn't know that that would be a part of that effort.  👍